config/lib/seed.js

```diff
@@ -2,87 +2,39 @@
 
 var mongoose = require('mongoose'),
   chalk = require('chalk'),
-  crypto = require('crypto'),
-  User = mongoose.model('User');
-
-console.log(chalk.bold.red('Warning:  Database seeding is turned on'));
-
-var seedUser = {
-  username: 'user',
-  password: 'User_Password1!',
-  provider: 'local',
-  email: 'user@localhost.com',
-  firstName: 'User',
-  lastName: 'Local',
-  displayName: 'User Local',
-  roles: ['user']
-};
-
-var seedAdmin = {
-  username: 'admin',
-  password: 'Admin_Password1!',
-  provider: 'local',
-  email: 'admin@localhost.com',
-  firstName: 'Admin',
-  lastName: 'Local',
-  displayName: 'Admin Local',
-  roles: ['user', 'admin']
-};
-
-//If production only seed admin if it does not exist
-if (process.env.NODE_ENV === 'production') {
-  //Add Local Admin
-  User.find({username: seedAdmin.username}, function (err, users) {
-    if (users.length === 0) {
-      var user = new User(seedAdmin);
-
-      // generate a random password and save 
-      User.generateRandomPassphrase()
-      .then(saveUser(user))
-      .catch(reportError);
+  crypto = require('crypto');
 
-    } else {
-      console.log(seedAdmin.username + ' user exists');
-    }
-  });
-} else {
+// save the specified user with the password provided from the resolved promise
+function saveUser(user, password) {
+  return new Promise(function (resolve, reject) {
 
-  //Add Local User
-  User.find({username: seedUser.username}).remove(function () {
-    var user = new User(seedUser);
+    var User = mongoose.model('User');
 
-    // generate a random password and save
-    User.generateRandomPassphrase()
-    .then(saveUser(user))
-    .catch(reportError);
-  });
+    User.find({username: user.username}).remove(function (err, users) {
 
-  //Add Local Admin
-  User.find({username: seedAdmin.username}).remove(function () {
-    var user = new User(seedAdmin);
+      if (err) {
+        reject(new Error('Database Seeding:\t\t\tFailed to add local ' + user.username));
+      }
 
-    // generate a random password and save
-    User.generateRandomPassphrase()
-    .then(saveUser(user))
-    .catch(reportError);
-  });
-}
+      // For admin user account we do not remove them
+      if (user.username === 'admin' && users.length === 0) {
```

Comment:
I think this should be checking for a length greater than zero, or just `users.length`

Even with that fix, this still won't work. We're already in the callback of the `remove()`, so it's too late to check if we should be removing this user.

I think it might be a good idea to move the remove functionality outside of the `saveUser` method. We probably don't want to enter this function unless we are sure we want to add the new user.